### DISCUSSION TITLE
Service Info API sever decoupling: a discussion

### DISCUSSION THREAD

**Discussion by 7flying** [fdo-rs/fido-device-onboard-rs#572] [Ideas]
This issue is simply to discuss the pros and cons of having the Service Info API server implementation decoupled from the Owner Onboarding server.
Should we merge them? Should we leave them as they are?

**Comment by miabbott** [comment]
Instead of an issue, let's just convert this to a discussion topic?

There's a handy `Convert to discussion` link at the bottom right

**Comment by miabbott** [comment]
@mmartinv I believe you raised this issue originally; could you share your thoughts/questions?

**Comment by mmartinv** [comment]
According to the FIDO Onboard specification, the _ServiceInfo_ messages are exchanged between the Device and the Owner's onboarding server and is the way to customize/configure the Device and its Management Agent. This interaction occurs only once in the device lifetime, during the final stage of the Transfer of Ownership 2 (TO2) protocol.

The implementation of this part of the specification was moved to the _ServiceInfo API Server_ by defining a non-standard REST API between the Owner's onboarding server and the _ServiceInfo API Server_. From my point of view this is hard to justify and adds complexity to the implementation with little/no benefits.

The implementation of additional FSIMs will require big changes in the _ServiceInfo API Server_ to allow the bidirectional exchange of messages so I think this is the chance to merge the _ServiceInfo API Server_ into Owner's onboarding server and simplify things.

**Comment by 7flying** [comment]
Merging service info API server with Owner onboarding server:

PROS
- Adds less complexity
- One less API to maintain 

CONS
- Less scalable (?¿)
  - Related to reading config: Not really since we will be using a DB
  - Related to performance: the devices are still connecting to the owner as they were before, there weren’t two separate endpoints to connect based on the request type.
- At the time of writing, requires restart for configuration
  - Minor: fswatcher(s) in rust are easy https://github.com/notify-rs/notify 

The team agrees on the merge, see issue #622
